src/main/java/io/apicurio/datamodels/transform/OpenApi30to31TransformationVisitor.java

```diff
@@ -52,23 +54,29 @@ public OpenApi31Document getResult() {
         return this.doc31;
     }
 
+    @Override
+    public void setTraversalContext(TraversalContext context) {
+        this.traversalContext = context;
+    }
+
     /**
      * @see io.apicurio.datamodels.models.visitors.Visitor#visitSchema(io.apicurio.datamodels.core.models.common.Schema)
      */
     @Override
     public void visitSchema(Schema node) {
-        mapSchema((OpenApi30Schema) node, findMatchingNode(node));
+        mapSchema((OpenApi30Schema) node, findMatchingNode());
     }
 
-    private OpenApi31Schema findMatchingNode(Schema node) {
-        NodePath nodePath = NodePathUtil.createNodePath(node);
```

Comment:
`NodePathUtil.createNodePath(node)` walks the entire tree (very slow for the large GitHub API in the codegen project). This is just an optimization to use the `TraversalContext` at the point the schema is visited.